public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -123,21 +156,121 @@
+                            .domainId(cdr.getDomainId())
+                            .domainDisplay(cdr.getDomainDisplay())
+                            .domainDesc(cdr.getDomainDesc())
+                            .dbType(cdr.getDbType())
+                            .domainRoute(cdr.getDomainRoute())
+                            .conceptId(cdr.getConceptId())
+                            .countValue(cdr.getCountValue());
+                }
+            };
+
+
+
+
 
+    /**
+     * Converter function from backend representation (used with Hibernate) to
+     * client representation (generated by Swagger).
+     */
+    private static final Function<AchillesResult, org.pmiops.workbench.model.AchillesResult>
+            TO_CLIENT_ACHILLES_RESULT =
+            new Function<AchillesResult, org.pmiops.workbench.model.AchillesResult>() {
+                @Override
+                public org.pmiops.workbench.model.AchillesResult apply(AchillesResult o) {
+
+                    return new org.pmiops.workbench.model.AchillesResult()
+                            .id(o.getId())
+                            .analysisId(o.getAnalysisId())
+                            .stratum1(o.getStratum1())
+                            .stratum2(o.getStratum2())
+                            .stratum3(o.getStratum3())
+                            .stratum4(o.getStratum4())
+                            .stratum5(o.getStratum5())
+                            .stratum5Name(o.getStratum5Name())
+                            .countValue(o.getCountValue());
                 }
             };
 
 
+    /* Final api functions put on top */
+    @Override
+    public ResponseEntity<DbDomainListResponse> getDomainFilters() {
+        List<DbDomain> domains=dbDomainDao.findByDbType("domain_filter");
+        DbDomainListResponse resp=new DbDomainListResponse();
+        resp.setItems(domains.stream().map(TO_CLIENT_DBDOMAIN).collect(Collectors.toList()));
+        return ResponseEntity.ok(resp);
+    }
+
+    @Override
+    public ResponseEntity<DbDomainListResponse> getSurveyList() {
+        List<DbDomain> domains=dbDomainDao.findByDbTypeAndAndConceptIdNotNull("survey");
+        DbDomainListResponse resp=new DbDomainListResponse();
+        resp.setItems(domains.stream().map(TO_CLIENT_DBDOMAIN).collect(Collectors.toList()));
+        return ResponseEntity.ok(resp);
+    }
+
+
+
+    @Override
+    public ResponseEntity<QuestionConceptListResponse> getSurveyResults(String surveyConceptId) {
+        /* Set up the age and gender names */
+        // Too slow and concept names wrong so we hardcode list
+        // List<Concept> genders = conceptDao.findByConceptClassId("Gender");
+        this.setAgeStratumNameMap();
+        this.setGenderStratumNameMap();
+        long longSurveyConceptId = Long.parseLong(surveyConceptId);
+        List<QuestionConcept> questions = questionConceptDao.findSurveyQuestions(longSurveyConceptId);
```

Comment:
So... the client passes you a concept ID and you get the questions that have this as an answer? How does this get used?

Reply:
This is for results of entire survey. Client passes a survey concept id and we get all the questions and analyses with results for that surevy. 

We will also have a search where results are returned for only questions matching the search results.